carma_wm/src/WMListenerWorker.cpp

```diff
@@ -52,8 +52,21 @@ void WMListenerWorker::mapCallback(const autoware_lanelet2_msgs::MapBinConstPtr&
     map_callback_();
   }
 }
+
+bool WMListenerWorker::checkIfReRoutingNeeded()
+{
+  return set_flag_;
+}
+
 void WMListenerWorker::mapUpdateCallback(const autoware_lanelet2_msgs::MapBinConstPtr& geofence_msg) const
 {
+
+  if(geofence_msg.invalidates_route==true)
+  {
+  local_geofence_msg_=geofence_msg;
+  set_flag_=true;
+  }
```

Comment:
This is supposed to not apply the map until the route is received. You need to return in that if call so that the rest of this method does not run.